risc0/build/risc0.ld

```diff
@@ -21,10 +21,9 @@ EXTERN(__start)
 
 /* Must match risc0/zkvm/platform/src/memory.rs */
 MEMORY {
-  stack        : ORIGIN = 0x00000000, LENGTH =  16M
-  data    (RW) : ORIGIN = 0x01000000, LENGTH =  16M
-  heap         : ORIGIN = 0x02000000, LENGTH =  32M
-  input        : ORIGIN = 0x04000000, LENGTH =  32M
+  stack        : ORIGIN = 0x02000000, LENGTH =  16M
+  data    (RW) : ORIGIN = 0x03000000, LENGTH =  16M
+  heap         : ORIGIN = 0x04000000, LENGTH =  32M
   prog    (X)  : ORIGIN = 0x06000000, LENGTH =  32M
```

Comment:
Is there a reason not to put prog @ 0x08... - 0x0c...?  I was of the impression that system memory starts only after 0x0c and that nondet was now below @0x02.  It seems like 64M for heap and 64M for code would be better than 32M for heap and 96M for program.  Or maybe even more room for heap.

Reply:
That all makes great sense. I was also thinking maybe `prog` should live above the system region? Like maybe reading from the PC could use a special memory op.